### PR TITLE
AuthN: feature flag for external service authentication

### PR DESCRIPTION
**Pull request by gamab**
**What is this feature?**

The AuthNZ squad is working on providing an OAuth2 server that would allow external services (ex: plugins) to authenticate against Grafana using the standard OAuth2 flows.

This PR introduces a new feature flag, that will allow us to start merging our work.

**Why do we need this feature?**

[Add a description of the problem the feature is trying to solve.]

**Who is this feature for?**

[Add information on what kind of user the feature is for.]

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [x] It works as expected from a user's perspective.
- [x] If this is a pre-GA feature, it is behind a feature toggle.
- [x] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Review by chri2547 — APPROVED** [review]

**Review by andresmgot — APPROVED** [review]
LGTM, you need to merge the latest `main`

File: docs/sources/setup-grafana/configure-grafana/feature-toggles/index.md [hunk 1]
```diff
@@ -112,8 +112,9 @@ Alpha features might be changed or removed without prior notice.
...
-| ------------------- | --------------------------------------------------- |
-| `k8s`               | Explore native k8s integrations                     |
-| `entityStore`       | SQL-based entity store (requires storage flag also) |
-| `nestedFolders`     | Enable folder nesting                               |
+| Feature toggle name   | Description                                                    |
+| --------------------- | -------------------------------------------------------------- |
+| `k8s`                 | Explore native k8s integrations                                |
+| `entityStore`         | SQL-based entity store (requires storage flag also)            |
+| `nestedFolders`       | Enable folder nesting                                          |
+| `externalServiceAuth` | Starts an OAuth2 authentication provider for external services |
```
Comment (mgyongyosi) [inline docs/sources/setup-grafana/configure-grafana/feature-toggles/index.md:120]: Can we rephrase it to say something like "Enables an OAuth2 authentication provider for external services"?

I think we should also introduce a config parameter eventually which specifies whether the OAuth2 server is turned on or not and those two (the config and the ff) should work jointly. Wdyt?
